includes/services/extended/0-google-base.php

```diff
@@ -29,9 +29,11 @@ function __construct() {
 		$this->set_endpoint( 'userinfo', 'https://www.googleapis.com/oauth2/v3/userinfo', 'GET' );
 
 		$creds              = $this->get_credentials();
-		$this->redirect_uri = $creds['redirect_uri'];
-		$this->key          = $creds['key'];
-		$this->secret       = $creds['secret'];
+		if ( ! empty( $creds ) ) {
```

Comment:
This looks good. I'm surprised we didn't get this warning before 7.4.

As this is in so many of the services, should we take this opportunity to refactor it and have something like a `init_credentials` method in the base class?

Reply:
I was thinking the same, but felt like that could be a slippery slope since there's a lot of duplication.  But I think moving this into a single function that you named is a good change to include here. I'll do that and push shortly.

Reply:
Added in a2a35f9201718c5e82550b1001f6fc5c32cd38ce :rocket: